server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3378,6 +3384,63 @@ private Pair<List<ServiceOfferingJoinVO>, Integer> searchForServiceOfferingsInte
         return _srvOfferingJoinDao.searchAndCount(sc, searchFilter);
     }
 
+    /**
+     * Retrieves the API caller. If the projectId or accountName parameters were provided in the API call, consider the account or project as the API caller.
+     * With that, the APIs return will be filtered by the account defined in the API call.
+     * @param originalCaller Account that made the API call
+     * @param projectId projectId provided in API call
+     * @param accountName accountName provided in API call
+     * @param domainId domainId provided in API call
+     * @return Account object
+     */
+    protected Account getCallerAccordingToProjectIdAndAccountNameAndDomainId(Account originalCaller, Long projectId, String accountName, Long domainId) {
```

Comment:
@RodrigoDLopez 
I think these new methods can be replaced by _accountMgr.finalizeOwner
please try with it.